scipy/interpolate/fitpack2.py

```diff
@@ -172,6 +172,8 @@ def __init__(self, x, y, w=None, bbox=[None]*2, k=3, s=None,
             if (not np.isfinite(x).all() or not np.isfinite(y).all() or
                     not w_finite):
                 raise ValueError("x and y array must not contain NaNs or infs.")
+        if not all(diff(x) > 0.0):
+            raise TypeError('x must be strictly increasing')
```

Comment:
This should be ``ValueError`` not ``TypeError`` (data does have the right type, but values have an issue - nonmonotonicity). I see that you used ``TypeError`` for consistency with other checks already present in this file - could you change those to ``ValueError`` as well?